Unable to import a package that carries the current package name in its path

The following program `sample.go` triggers a panic:

```go
package foo

import "github.com/foo/app"

func Bur() {
	app.Bar()
}
```

Expected result:
```console
$ go run sample.go 
sample.go:3:8: cannot find package "github.com/foo/app" in any of:
	/usr/lib/go/src/github.com/foo/app (from $GOROOT)
	/home/jlevesy/code/src/github.com/foo/app (from $GOPATH)
```

Got:
```console
$ yaegi ./sample.go
runtime error: index out of range [-1]
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0003a4000, 0x40)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc000299df0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:330 +0xbc
panic(0xd72380, 0xc0000aa0c0)
	/usr/lib/go/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.effectivePkg(0xc0000a8070, 0x3, 0xc0000aa081, 0x12, 0xfac5c0, 0xc000394390)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/src.go:196 +0x660
github.com/containous/yaegi/interp.pkgDir(0xc00003a027, 0x12, 0xc0000a8070, 0x3, 0xc0000aa081, 0x12, 0x3, 0xc000020000, 0xcf7900, 0xd28a20, ...)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/src.go:151 +0x1d1
github.com/containous/yaegi/interp.(*Interpreter).importSrc(0xc0000ce1a0, 0xc0000a8070, 0x3, 0xc0000aa081, 0x12, 0x0, 0xc000092601, 0xc000299688, 0x540161)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/src.go:29 +0x1412
github.com/containous/yaegi/interp.(*Interpreter).gta.func1(0xc0000b4400, 0x7ffc12226195)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/gta.go:178 +0x112b
github.com/containous/yaegi/interp.(*node).Walk(0xc0000b4400, 0xc000299bd0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:188 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc0000b4300, 0xc000299bd0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:192 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000b4100, 0xc000299bd0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:192 +0x70
github.com/containous/yaegi/interp.(*Interpreter).gta(0xc0000ce1a0, 0xc0000b4100, 0xc0000a8070, 0x3, 0x7ffc12226195, 0x9, 0x0, 0xc0000ac100, 0x0, 0x0, ...)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/gta.go:14 +0x131
github.com/containous/yaegi/interp.(*Interpreter).gtaRetry(0xc0000ce1a0, 0xc000299d98, 0x1, 0x1, 0xc0000a8070, 0x3, 0x7ffc12226195, 0x9, 0x0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/gta.go:239 +0x14a
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000ce1a0, 0xc0000ba050, 0x44, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:352 +0x237
main.main()
	/home/jlevesy/code/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:143 +0x553
```

--- a/interp/src.go
+++ b/interp/src.go
@@ -191,7 +191,8 @@ func effectivePkg(root, path string) string {
 	for i := 0; i < len(splitPath); i++ {
 		part := splitPath[len(splitPath)-1-i]
 
-		if part == splitRoot[len(splitRoot)-1-rootIndex] && i != 0 {
+		index := len(splitRoot) - 1 - rootIndex
+		if index > 0 && part == splitRoot[index] && i != 0 {
 			prevRootIndex = rootIndex
 			rootIndex++
 		} else if prevRootIndex == rootIndex {

--- a/interp/src_test.go
+++ b/interp/src_test.go
@@ -26,6 +26,12 @@ func Test_effectivePkg(t *testing.T) {
 			path:     "vendor/guthib.com/containous/vin",
 			expected: "github.com/foo/plugin/vendor/guthib.com/containous/fromage/vendor/guthib.com/containous/vin",
 		},
+		{
+			desc:     "path is non-existent",
+			root:     "foo",
+			path:     "githib.com/foo/app",
+			expected: "foo/githib.com/foo/app",
+		},
 	}
 
 	for _, test := range testCases {

